vue-app/src/components/Cart.vue

```diff
@@ -6,12 +6,22 @@
       <wallet-widget :isActionButton="true" />
     </div>
     <div v-else class="cart-container">
-      <div class="reallocation-message" v-if="$store.getters.canUserReallocate">
+      <div
+        class="reallocation-message"
+        v-if="$store.getters.canUserReallocate && $store.getters.hasUserVoted"
```

Comment:
Won't `hasUserVoted` always be true if `canUserReallocate` is true?

Reply:
If not, perhaps we should update `canUserReallocate` logic to ensure the user has already voted?

Reply:
I think so - a user must vote in order to finish their contribution. Before then I think `canUserReallocate` should be false.

Reply:
Totally!

This is what I did in the first place.
```ts
canUserReallocate: (_, getters): boolean => {
  return (
    getters.hasUserContributed &&
    getters.hasUserVoted && // <--- to be able to reallocate, the user must vote first
    (getters.isRoundContributionPhase || getters.isRoundReallocationPhase)
  )
}
```

The problem I had with this approach (which I think is the best because it is clearer to future developers) is that the number of things I needed to change in the Cart UI logic was big and complex. Mainly because we want to display the same reallocation UI but with this flag off. There are a lot of conditions around `canUserReallocate`.

So, I'm keeping `canUserReallocate` truthy because it is easier from the refactoring perspective. I only needed to add a couple of conditions to the Cart logic using `hasUserVoted`.

I hate this because we are adding more and more conditions in the Cart logic what makes it super difficult to understand and maintain. I think this is a task we could tackle in the Technical Debt epic! :)